lib/pages/class_feedback/service/feedback_provider.dart

```diff
@@ -0,0 +1,147 @@
+      final List<dynamic> options = json['options'];
+      final List<String> optionsString =
+          options.map((e) => e[LocaleProvider.localeString] as String).toList();
+      return FeedbackQuestionDropdown(
+        category: json['category'],
+        question: json['question'][LocaleProvider.localeString],
+        id: id,
+        answerOptions: optionsString,
+      );
+    } else if (json['type'] == 'rating') {
+      return FeedbackQuestionRating(
+        category: json['category'],
+        question: json['question'][LocaleProvider.localeString],
+        id: id,
+      );
+    } else if (json['type'] == 'text') {
+      return FeedbackQuestionText(
+        category: json['category'],
+        question: json['question'][LocaleProvider.localeString],
+        id: id,
+      );
+    } else if (json['type'] == 'slider') {
+      return FeedbackQuestionSlider(
+        category: json['category'],
+        question: json['question'][LocaleProvider.localeString],
+        id: id,
+      );
+    } else {
+      return FeedbackQuestion(
+        category: json['category'],
+        question: json['question'][LocaleProvider.localeString],
+        id: id,
+      );
+    }
+  }
+}
+
+class FeedbackProvider with ChangeNotifier {
+  Future<bool> addResponse(FeedbackQuestionAnswer response) async {
+    try {
+      await FirebaseFirestore.instance
+          .collection('forms')
+          .doc('class_feedback_answers')
+          .collection(response.questionNumber)
+          .add(response.toData());
+      return true;
+    } catch (e) {
+      print(e);
+      AppToast.show(S.current.errorSomethingWentWrong);
+      return false;
+    }
+  }
+
+  Future<Map<String, FeedbackQuestion>> fetchQuestions() async {
+    try {
+      final DocumentSnapshot documentSnapshot = await FirebaseFirestore.instance
+          .collection('forms')
+          .doc('class_feedback_questions')
+          .get();
+      final Map<String, dynamic> data = documentSnapshot['questions'];
+      final Map<String, FeedbackQuestion> questions = {};
+      for (final value in data.values) {
+        final key = data.keys.firstWhere((element) => data[element] == value);
+        questions[key] = FeedbackQuestionExtension.fromJSON(value, key);
+      }
+      return questions;
+    } catch (e) {
+      print(e);
+      AppToast.show(S.current.errorSomethingWentWrong);
+      return null;
+    }
+  }
+
+  Future<Map<String, dynamic>> fetchCategories() async {
+    try {
+      final DocumentSnapshot documentSnapshot = await FirebaseFirestore.instance
+          .collection('forms')
+          .doc('class_feedback_questions')
+          .get();
+      final Map<String, dynamic> data = documentSnapshot['categories'];
+      return data;
+    } catch (e) {
+      print(e);
+      AppToast.show(S.current.errorSomethingWentWrong);
+      return null;
+    }
+  }
+
+  Future<bool> setUserClassFeedback(String className, String uid) async {
```

Comment:
It's not immediately clear from the name what this method does. Maybe `setUserSubmittedFeedbackForClass` or something? Also, I think there should be a method like `submitFeedback` that does `addResponse` and then calls this method - don't rely too much on the UI module taking care of the business logic. These methods can then be private here..